src/Discord.Net.SlashCommands/SlashCommandService.cs

```diff
@@ -0,0 +1,675 @@
+        {
+            EnsureClientReady();
+
+            var props = _typedModuleDefs.Values.SelectMany(x => x.ToApplicationCommandProps()).ToList();
+
+            if (!deleteMissing)
+            {
+                var existing = await ClientHelper.GetGuildApplicationCommands(Client, guildId).ConfigureAwait(false);
+                var missing = existing.Where(x => !props.Any(y => y.Name.IsSpecified && y.Name.Value == x.Name));
+                props.AddRange(missing.Select(x => x.ToApplicationCommandProps()));
+            }
+
+            return await ClientHelper.BulkOverwriteGuildApplicationCommand(Client, guildId, props.ToArray()).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Register/update the Application Commands to Discord
+        /// </summary>
+        /// <param name="deleteMissing">If true, delete all of the commands that are not registered in the <see cref="SlashCommandService"/></param>
+        /// <returns>A task representing the command registration process, with a collection of <see cref="RestGlobalCommand"/> containing the
+        /// global commands that are currently registered to the Discord
+        /// </returns>
+        public async Task<IReadOnlyCollection<RestGlobalCommand>> RegisterCommandsGloballyAsync (bool deleteMissing = true)
+        {
+            EnsureClientReady();
+
+            var props = _typedModuleDefs.Values.SelectMany(x => x.ToApplicationCommandProps()).ToList();
+            
+            if (!deleteMissing)
+            {
+                var existing = await ClientHelper.GetGlobalApplicationCommands(Client).ConfigureAwait(false);
+                var missing = existing.Where(x => !props.Any(y => y.Name.IsSpecified && y.Name.Value == x.Name));
+                props.AddRange(missing.Select(x => x.ToApplicationCommandProps()));
+            }
+
+            return await ClientHelper.BulkOverwriteGlobalApplicationCommand(Client, props.ToArray()).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Register a set of commands as "Guild Commands" to a guild
+        /// </summary>
+        /// <remarks>
+        /// Commands will be registered as standalone commands, if you want the <see cref="SlashGroupAttribute"/> to take effect,
+        /// use <see cref="AddModulesToGuild(IGuild, ModuleInfo[])"/>
+        /// </remarks>
+        /// <param name="guild">Guild the commands will be registered to</param>
+        /// <param name="commands">Commands that will be registered</param>
+        /// <returns>A task representing the command registration process, with a collection of <see cref="RestGuildCommand"/> containing the
+        /// commands that are currently registered to the provided guild as its result.
+        /// </returns>
+        public async Task<IReadOnlyCollection<RestGuildCommand>> AddCommandsToGuildAsync (IGuild guild, params IApplicationCommandInfo[] commands)
```

Comment:
Maybe make this accept a module definition to allow registering of a specific module. this function seems redundant as the client has this option

Reply:
I added a AddModulesToGuildAsync method